### PR TITLE
refactor: finally remove async_stream

### PR DESCRIPTION
**Pull request by CHr15F0x**
Depends on #2152

### CODE REVIEW

File: crates/pathfinder/src/sync/class_definitions.rs [hunk 1]
```diff
@@ -95,13 +96,13 @@ pub(super) fn declared_class_counts_stream(
 ) -> impl futures::Stream<Item = anyhow::Result<usize>> {
     const BATCH_SIZE: usize = 1000;
 
-    async_stream::try_stream! {
+    let (tx, rx) = mpsc::channel(1);
+    thread::spawn(move || {
```
[the remark this answers is not in the document]
Comment (kkovaacs) [reply crates/pathfinder/src/sync/class_definitions.rs:100]: Oh, this is _not_ in a request handler but in checkpoint sync -- so it's obviously limited by how many instances of this we instantiate.

File: crates/pathfinder/src/sync/class_definitions.rs [hunk 2]
```diff
@@ -95,13 +96,13 @@ pub(super) fn declared_class_counts_stream(
...
         let mut batch = Vec::<usize>::new();
 
         while start <= stop {
             if let Some(counts) = batch.pop() {
-                yield counts;
-                continue;
+                _ = tx.blocking_send(Ok(counts));
```
[the remark this answers is not in the document]
Comment (CHr15F0x) [reply crates/pathfinder/src/sync/class_definitions.rs:105]: Yeah it didn't make any sense.

[the remark this answers is not in the document]
Comment (CHr15F0x) [reply crates/pathfinder/src/sync/class_definitions.rs:105]: You've just discovered a regression, thanks!

[the remark this answers is not in the document]
Comment (CHr15F0x) [reply crates/pathfinder/src/sync/class_definitions.rs]: [0ce7f10](https://github.com/eqlabs/pathfinder/pull/2164/commits/0ce7f10b6d0a0ee7ee131045290fc12f0b4cd4a5)

**Pull request merged**